packages/uicore/src/components/ScrollContainer/ScrollContainer.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { CSSProperties, FC, useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from 'react'
+import cx from 'classnames'
+import rafSchd from 'raf-schd'
+import css from './ScrollContainer.css'
+
+export interface ScrollContainerProps {
+  /**
+   * `true`: renders top shadow when required.
+   *
+   * `false`: never renders top shadow.
+   *
+   * default: `true` */
+  showTopShadow?: boolean
+  /**
+   * `true`: renders bottom shadow when required
+   *
+   * `false`: never renders bottom shadow
+   *
+   * default: `true` */
+  showBottomShadow?: boolean
+  /**
+   * height of the container
+   *
+   * default: `100%` */
+  height?: CSSProperties['height']
+  className?: string
+  overflowWrapperClassName?: string
+  testId?: string
+}
+
+export const ScrollContainer: FC<ScrollContainerProps> = props => {
+  const {
+    children,
+    className,
+    overflowWrapperClassName,
+    testId,
+    height = '100%',
+    showBottomShadow = true,
+    showTopShadow = true
+  } = props
+  const [scrollShadows, setScrollShadows] = useState({ top: false, bottom: false })
+  const scrollableRef = useRef<HTMLDivElement>(null)
+  const resizeObserverRef = useRef<ResizeObserver>()
+
+  const setShadows = useCallback(
+    rafSchd(() => {
+      const target = scrollableRef.current
+
+      if (!target) return
+
+      const { clientHeight, scrollHeight, scrollTop } = target
+      const hasScrollableContent = scrollHeight > clientHeight
+
+      if (!hasScrollableContent) return setScrollShadows({ top: false, bottom: false })
+
+      const isTop = scrollTop === 0
+      const isBetween = scrollTop > 0 && scrollTop < scrollHeight - clientHeight
+      const isBottom = scrollTop === scrollHeight - clientHeight
+
+      if (isTop) {
+        setScrollShadows({ top: false, bottom: true })
+      } else if (isBetween) {
+        setScrollShadows({ top: true, bottom: true })
+      } else if (isBottom) {
+        setScrollShadows({ top: true, bottom: false })
+      }
+    }),
+    []
+  )
+
+  useEffect(() => {
+    if (scrollableRef.current) {
+      scrollableRef.current.addEventListener('scroll', setShadows)
+
+      resizeObserverRef.current = new ResizeObserver(() => {
+        setShadows()
+      })
```

Comment:
No problem with this... just curious if it's possible to eliminate the extra function. Totally fine as-is
```suggestion
      resizeObserverRef.current = new ResizeObserver(setShadows)
```